core/pta/tcta.c

```diff
@@ -17,8 +53,32 @@ uint32_t trustcore_smc_handler(uint8_t sub_id, uint64_t arg1,
     EMSG("arg1 0x%"PRIx64, arg1);
     EMSG("arg2 0x%"PRIx64, arg2);
     EMSG("arg3 0x%"PRIx64, arg3);
+    
+    /* Virtual address of the shared memory */
+    vaddr_t *va;
+    tc_shmem_t *tct_shm;
 
+    switch(sub_id)
+    {
+	    case TC_SHMEM_TEE_SUB_ID:
+		    va = (vaddr_t)core_mmu_add_mapping(MEM_AREA_RAM_NSEC,arg1,sizeof(tc_shmem_t));
+		    EMSG("VA 0x%" PRIx64 " ", va);
+		    if (!va)
+                        goto bad_addr;
```

Comment:
goto is not really required here. Just return if the va is not NULL

Reply:
Done.  Updated it.